wallet/src/de/schildbach/wallet/ui/TransactionResultActivity.kt

```diff
@@ -77,10 +83,16 @@ class TransactionResultActivity : AbstractWalletActivity() {
             transactionResultViewBinder.bind(tx, payeeName, payeeVerifiedBy)
             view_on_explorer.setOnClickListener { viewOnExplorer(tx) }
             transaction_close_btn.setOnClickListener {
-                if (intent.getBooleanExtra(EXTRA_USER_AUTHORIZED_RESULT_EXTRA, false)) {
-                    startActivity(WalletActivity.createIntent(this))
-                } else {
-                    startActivity(LockScreenActivity.createIntentAsNewTask(this))
+                when {
+                    intent.action == Intent.ACTION_VIEW -> {
+                        finish()
+                    }
+                    intent.getBooleanExtra(EXTRA_USER_AUTHORIZED_RESULT_EXTRA, false) -> {
+                        startActivity(WalletActivity.createIntent(this))
+                    }
+                    else -> {
+                        startActivity(LockScreenActivity.createIntentAsNewTask(this))
+                    }
```

Comment:
This part doesn't seem to work.

When using the secondary app that makes a payment request, the last screen of the payment process is the Transaction Result screen with a "Close" button.  Clicking the close button appears to execute the "else" clause here rather than the first clause that will **finish()** the activity.

Reply:
As we discussed on Slack the left button of the "Hack" app works in a hacky way so you should use the right button,

Reply:
Yes, this code is working under normal circumstances.